### ISSUE TITLE
Import playlist when opening playlist with Mixxx

### ISSUE THREAD

**Issue by KristjanESPERANTO** [mixxxdj/mixxx#11178]
### Feature Description

The following feature would be very helpful: Mixx directly imports playlists that I open via a file manager.

Workflow: any file manager -> right-click on playlist (*.m3u) -> open with Mixxx -> Mixxx opens and imports the playlist

By the way: Thanks for this great software! 👏

**Comment by daschuer** [comment]
Thank you for the feature request. This one is now waiting for a contributor who has interest to implement it.

**Comment by harsv689** [comment]
Sir, can i work on this ?

**Comment by KristjanESPERANTO** [comment]
@harsv689 So far no one has reported working on the problem. I would be happy if you take care of it.

**Comment by shivang0-0** [comment]
I would like to work on this issue if it's still unassigned.

**Comment by daschuer** [comment]
Go ahead. Do you need help?

**Comment by shivang0-0** [comment]
> Go ahead. Do you need help?

Yes please, I was going through the following parser code [here](https://github.com/mixxxdj/mixxx/blob/main/src/util/cmdlineargs.cpp#L266-L270).

I guess, I do not need to add any extra positional argument like "playlist". The "file" positional argument can be used and in [coreservices.cpp](https://github.com/mixxxdj/mixxx/blob/3d1f82b1f3c7f9fab4268ef11ab53ce571ef27f7/src/coreservices.cpp#L473) an else statement can be introduced for files with extension .m3u and then `void BasePlaylistFeature::slotImportPlaylistFile(const QString& playlist_file)` [here](https://github.com/mixxxdj/mixxx/blob/main/src/library/trackset/baseplaylistfeature.cpp#L475) can be used to import the playlist. Correct me, if I'm wrong in the above workflow please.

Moreover, while testing these changes out, is there a way faster that building the project again using cmake, like for verifying small changes?

**Comment by ronso0** [comment]
> an else statement can be introduced for files with extension .m3u

Let's support all extensions that can be imported.
Just open a PR and the code be discussed much easier.

**Comment by shivang0-0** [comment]
> > an else statement can be introduced for files with extension .m3u
> 
> Let's support all extensions that can be imported. Just open a PR and the code be discussed much easier.

But, isn't the extension of all the exported playlist that needs to be imported is .m3u, so I was trying to identify .m3u files in particular. Any heads up, will be much appreciated on this @daschuer and @ronso0.

**Comment by ronso0** [comment]
Afaik Mixxx supports m3u, m3u8 and pls.
Just right-click on the Playlist feature and test what you can import.

**Comment by shivang0-0** [comment]
will 

> Afaik Mixxx supports m3u, m3u8 and pls. Just right-click on the Playlist feature and test what you can import.

will do a check around then, thanks @ronso0 !!

**Comment by shivang0-0** [comment]
@daschuer I've change certain aspects of the original `void BasePlaylistFeature::slotCreateImportPlaylist()` function to make it independent of the qAction but when calling this new function with a playlist file's path as an argument. I get this as an error "qualified name is not allowed". If possible can you review the code once?

**Comment by ronso0** [comment]
> Just open a PR and the code be discussed much easier.

**Comment by Nofates** [comment]
I am interested in the Gitpay task associated with this issue. Before coding, could a maintainer confirm that task 708 is still active and that a current-main PR from @anakinsuper remains eligible? My proposed scope is to accept an OS-opened playlist file and route all playlist extensions already supported by Mixxx (including .m3u, .m3u8, and .pls) through the existing import-to-Auto-DJ path, while avoiding duplicate existing work. I am an AI-assisted contributor operating under @anakinsuper; any submitted code will be reviewed and tested before submission. No payment details are included here.

**Comment by ronso0** [comment]
> I am interested in the Gitpay task associated with this issue

> No payment details are included here.

What are you referring to?
Mixxx is a fun project run by volunteers and there is no such thing as $ involved when contributing...

**Comment by Eve00000** [comment]
never heard about Gitpay before ...  OMG, are people expecting to be payed for contributions now?
